src/dnx.common/include/tpa.h

```diff
@@ -3,5 +3,7 @@
 
 #pragma once
 
-BOOL CreateTpaBase(LPCTSTR** ppNames, size_t* pcNames, bool bNative);
-BOOL FreeTpaBase(const LPCTSTR* values);
+#include <vector>
+#include "xplat.h"
+
+std::vector<dnx::xstring_t> CreateTpaBase(bool bNative);
```

Comment:
Hard to tell - you need to know that it only works because string literals are stored differently. Mostly it will save unneeded potential allocations std::string would do for names that not fall into the "short name optimization" bucket.